compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -978,18 +978,22 @@ impl Integrator<'_, '_> {
         new
     }
 
-    fn map_unwind(&self, unwind: Option<BasicBlock>) -> Option<BasicBlock> {
+    fn map_unwind(&self, unwind: UnwindAction) -> UnwindAction {
         if self.in_cleanup_block {
-            if unwind.is_some() {
-                bug!("cleanup on cleanup block");
+            match unwind {
+                UnwindAction::Cleanup(_) | UnwindAction::Continue => {
+                    bug!("cleanup on cleanup block");
+                }
+                UnwindAction::Unreachable | UnwindAction::Terminate => return unwind,
             }
-            return unwind;
         }
 
         match unwind {
-            Some(target) => Some(self.map_block(target)),
+            UnwindAction::Unreachable | UnwindAction::Terminate => unwind,
+            _ if let UnwindAction::Terminate = unwind => UnwindAction::Terminate,
```

Comment:
The condition is redundant with the arm before.

Reply:
Oh, this should be `_ if let UnwindAction::Terminate = self.cleanup_block => UnwindAction::Terminate,`